_posts/2018-08-14-apu2-validation-with-rte.md

```diff
@@ -0,0 +1,231 @@
+static IP set to `192.168.3.105`. To establish a connection, type:
+
+```
+ssh root@192.168.3.105
+```
+
+and login as root user - in my example I'm using Yocto RTE meta layer.
+Next open telnet connection to enable APU2 console preview. Check if ser2net
+redirection is configured:
+
+```
+root@orange-pi-zero:~# cat /etc/ser2net.conf
+13541:telnet:600:/dev/ttyS1:115200 8DATABITS NONE 1STOPBIT
+13542:telnet:600:/dev/ttyUSB0:115200 8DATABITS NONE 1STOPBIT
+```
+
+Open a new terminal or use `tmux` for split screen view and type:
+
+```
+telnet 192.168.3.105 13541
+```
+> we are using DB9 connector which is mapped to `/dev/ttyS1`
+
+Now, we have the ability to control our Device Under Test from our
+personal computer and watch APU2 console on another tab.
+
+To test, if our connections are correct, we can go through a firmware flashing
+procedure. I will use mainline release v4.8.0.2 from PCEngines [github](https://pcengines.github.io/)
+site. Generally, it's an easy task. First, power on DUT by typing:
+
+```
+root@orange-pi-zero:~# echo 1 > /sys/class/gpio/gpio199/value
+```
+
+Save the APU2 console boot log for firmware version validation. In my example
+it is:
+
+```
+PC Engines apu2
+coreboot build 20180608
+BIOS version v4.8.0.1
+4080 MB ECC DRAM
+```
+
+Send target firmware image via ssh, e.g.
+
+```
+scp apu2_v4.8.0.2.rom root@192.168.3.105:/tmp/coreboot.rom
+```
+
+Run commands manually or create a shell script:
+
+```sh
+#!/bin/bash
+
+# power on APU2 platform with relay
+echo 1 > /sys/class/gpio/gpio199/value
+sleep 1
+
+# force the APU platform into ACPI S5 state
+echo 1 > /sys/class/gpio/gpio410/value
+sleep 5
+echo 0 > /sys/class/gpio/gpio410/value
+
+# flash APU ROM with flashrom
+flashrom -f -p linux_spi:dev=/dev/spidev1.0,spispeed=16000 -w /tmp/coreboot.rom
+
+# power on APU2 board
+echo 1 > /sys/class/gpio/gpio410/value
+sleep 1
+echo 0 > /sys/class/gpio/gpio410/value
+```
+
+Example flashrom output after flash verification:
+
+```
+flashrom v0.9.9-r1955 on Linux 4.17.2 (armv7l)
+flashrom is free software, get the source code at https://flashrom.org
+
+Calibrating delay loop... OK.
+Found Winbond flash chip "W25Q64.V" (8192 kB, SPI) on linux_spi.
+Reading old flash chip contents... done.
+Erasing and writing flash chip... Erase/write done.
+Verifying flash... VERIFIED.
+```
+
+New APU2 console boot log:
+
+```
+PC Engines apu2
+coreboot build 20180705
+BIOS version v4.8.0.2
+4080 MB ECC DRAM
+```
+
+You can see that platform BIOS changed from `v4.8.0.1` to the newer version
+`v4.8.0.2`, which confirms that flashing with RTE was successful :)
+
+Above example is just one of multiple use cases that's available for RTE users.
+Our HAT has SPI, I2C, extended gpio and OC buffers headers, relay with power
+control capabilities, DB9 connector and 2 additional USB ports, all ready for
+personal use. Feel free and share your awesome setups in the comment section below!
```

Comment:
@artur-rs maybe this should be in the summary section, I would add more information about our plans for RTE eg. RPi version.